### ISSUE TITLE
Example from docs/fields.md fails for inline patchesStrategicMerge

### ISSUE THREAD

**Issue by TechnicalMercenary** [kubernetes-sigs/kustomize#1525]
I pasted the example yaml from `https://github.com/kubernetes-sigs/kustomize/blob/master/docs/fields.md#patchesstrategicmerge` into a Kustomization File and I'm getting an error...

My Kustomization FIle . ( simplified )

```
patchesStrategicMerge:
- |-
  apiVersion: apps/v1
  kind: Deployment
  metadata:
    name: nginx
  spec:
    template:
      spec:
        containers:
          - name: nginx
            image: nignx:latest
```

The Error....

```
[paxata-eks:us-west-2]$ kustomize build
Error: builtin patchStrategicMerge config: [112 97 116 104 115 58 10 45 32 124 45 10 32 32 97 112 105 86 101 114 115 105 111 110 58 32 97 112 112 115 47 118 49 10 32 32 107 105 110 100 58 32 68 101 112 108 111 121 109 101 110 116 10 32 32 109 101 116 97 100 97 116 97 58 10 32 32 32 32 110 97 109 101 58 32 110 103 105 110 120 10 32 32 115 112 101 99 58 10 32 32 32 32 116 101 109 112 108 97 116 101 58 10 32 32 32 32 32 32 115 112 101 99 58 10 32 32 32 32 32 32 32 32 99 111 110 116 97 105 110 101 114 115 58 10 32 32 32 32 32 32 32 32 32 32 45 32 110 97 109 101 58 32 110 103 105 110 120 10 32 32 32 32 32 32 32 32 32 32 32 32 105 109 97 103 101 58 32 110 105 103 110 120 58 108 97 116 101 115 116 10]: evalsymlink failure on '/Users/gregory/GitHub/devops-application/infrastructure/kustomize/resource/splunk/apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx
spec:
  template:
    spec:
      containers:
        - name: nginx
          image: nignx:latest' : lstat /Users/gregory/GitHub/devops-application/infrastructure/kustomize/resource/splunk/apiVersion: apps: no such file or directory
```

The version of Kustomize I'm using...

```
[paxata-eks:us-west-2]$ kustomize version
Version: {KustomizeVersion:3.1.0 GitCommit:95f3303493fdea243ae83b767978092396169baf BuildDate:2019-07-26T19:21:45+01:00 GoOs:darwin GoArch:amd64}
```

**Comment by Liujingfang1** [comment]
The inline patch is from HEAD, not available in 3.1.0.

**Comment by shakefu** [comment]
This issue is still valid... 4.5.7 still doesn't support this as far as I can tell.
